com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/validators/parameters/RoutesValidator.java

```diff
@@ -56,20 +57,19 @@ public Object attemptToCorrect(Object routes) {
         }
 
         return routesList.stream()
-            .map(route -> attemptToCorrectParameterMap(route))
+            .map(this::attemptToCorrectParameterMap)
             .collect(Collectors.toList());
     }
 
     private Map<String, Object> attemptToCorrectParameterMap(Map<String, Object> originalElem) {
-        Map<String, Object> correctedElem = new TreeMap<>();
-
-        for (String key : originalElem.keySet()) {
-            if (validators.containsKey(key)) {
-                correctedElem.put(key, attemptToCorrectParameter(validators.get(key), originalElem.get(key)));
-            }
-        }
+        return originalElem.entrySet()
+            .stream()
+            .filter(entry -> validators.containsKey(entry.getKey()))
+            .collect(TreeMap::new, this::collectMapEntries, Map::putAll);
```

Comment:
I'm still waiting for someone to tell me what are the benefits of using such collectors.